### PR TITLE
sched: Implement tkill/tgkill

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

https://linux.die.net/man/2/tgkill

## Impact

New API

## Testing

Pass sim:ostest

### CODE REVIEW

**Review by jerpelea — APPROVED** [review]

**Comment by acassis** [comment]
@xiaoxiang781216 why to remove pthread_kill() after adding tgkill()? Isn't it make our pthread incompatible with other POSIX OSes? Also take a look at PX5 RTOS: https://www.youtube.com/watch?v=6WO-Z32hKeE his justification to not using existent RTOS is because they are not pthread compatible with Linux. I think he doesn't know NuttX because our pthread is compatible with Linux.

**Comment by acassis** [comment]
@xiaoxiang781216 is it possible to keep pthread_kill() as a function that just call tgkill()? This way applications that needs pthread_kill() will still work on NuttX.

**Comment by xiaoxiang781216** [comment]
@acassis pthread_kill still exist, but move to libc:
https://github.com/apache/nuttx/pull/8224/files#diff-ed1c7fd078c8be1802e132c2ad837711277ddbae7c7cfe9e82236e0eebdb9839R60-R69
because since it's a simple wrapper around tkill.

**Comment by acassis** [comment]
Thank you @xiaoxiang781216! I didn't get the initial intention because the patch didn't show the move to libc, but in fact it was removed only as a syscall.

**Comment by xiaoxiang781216** [comment]
let's merge this patch? @acassis

File: sched/signal/sig_tgkill.c [hunk 1]
```diff
@@ -67,14 +65,13 @@
  *
  ****************************************************************************/
 
-int pthread_kill(pthread_t thread, int signo)
+int tgkill(pid_t pid, pid_t tid, int signo)
```
Comment (pkarashchenko) [inline sched/signal/sig_tgkill.c:68]: Do we need to add `UNUSED(pid);` or put 
```
#ifdef CONFIG_SCHED_HAVE_PARENT
  if (pid  != (pid_t)-1)
    {
      DEBUGASSERT(pid == rtcb->pid);
    }
#endif
```
?

Comment (xiaoxiang781216) [reply sched/signal/sig_tgkill.c:68]: Can not, since pid mean the main task pid, not the calling thread tid if it isn't -1.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**